RIOT-OS/RIOT
# boards/{bluepill*,blackpill*}: rename and improve doc

### Contribution description

- renamed:
    - `bluepill` --> `bluepill-stm32f103c8`
    - `bluepill-128kib` --> `bluepill-stm32f103cb`
    - `blackpill` --> `blackpill-stm32f103c8`
    - `blackpill-128kib` --> `blackpill-stm32f103cb`
- doc:
    - added link to corresponding hardware page on the STM32-base
      project
    - added picture

### Testing procedure

- check the documenation
- `make flash` and `make term` should still work for Bluepill and Blackpill boards (with updated values for `BOARD`, though)

### Issues/PRs references

None
